helixlauncher-core/src/meta.rs

```diff
@@ -0,0 +1,91 @@
+use std::io;
+
+use reqwest::Client;
+use thiserror::Error;
+
+use crate::config::Config;
+
+pub struct HelixLauncherMeta<'a> {
```

Comment:
I don't really like the name here, honestly... `helixlauncher_core::meta::HelixLauncherMeta`
Something like `MetaClient` or `ComponentMetaClient` could work?